### ISSUE TITLE
Pydantic alias keyword fiving `Alias name "<str>" is not a valid identifier` when it has spaces/dashes

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#9278]
After updating to 1.1.385 we started to see errors being reported when having pedantic field aliases with spaces or dashes. e.g.

```python
import pydantic


class SomeModel(pydantic.BaseModel):
    foo: str | None = pydantic.Field(
        default=None,
        alias="foo",
    )
    bar: str | None = pydantic.Field(
        default=None,
        alias="bar-alias",
    )
    baz: str | None = pydantic.Field(
        default=None,
        alias="baz-alias",
    )
```

Gives this:

```
/private/tmp/xxx/test.py
  /private/tmp/xxx/test.py:12:15 - error: Alias name "bar-alias" is not a valid identifier (reportGeneralTypeIssues)
  /private/tmp/xxx/test.py:17:15 - error: Alias name "baz-alias" is not a valid identifier (reportGeneralTypeIssues)
```

On 1.1.384 and before this was ok

**Comment by erictraut** [comment]
The new behavior is intended. See [this issue](https://github.com/microsoft/pyright/issues/9220).

**Comment by erictraut** [comment]
Reopening issue. We're revisiting this decision based on the discussion here: #9386.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.388.
